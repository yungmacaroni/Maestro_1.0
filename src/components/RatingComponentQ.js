import React, { useState } from "react";
import { View, TextInput, StyleSheet } from "react-native";
import { Picker } from "@react-native-picker/picker";
import { Player } from "./Player";

var players = {};

export default function RatingComponentQ(props) {
  const [playerName, setPlayerName] = useState("");
  const [dictKey, setDictKey] = useState("");
  const [abilityValue, setSelectedValue] = useState(null);

  function nameHandler(name) {
    delete players[dictKey];
    setPlayerName(name);
    setDictKey(name);
  }

  function abilityPicker(valueA) {
    delete players[dictKey];
    setSelectedValue(valueA);
  }

  function dataHandler(players) {
    props.onSubmit(players);
  }

  function dataCleaner(value) {
    props.setRefreshed(value)
  }

  if (dictKey != "" && abilityValue != null) {
    players[dictKey] = Player.QuickPlayer(playerName, abilityValue);
  }

  if (props.dataHandled == false) {
    dataHandler(players);
  }

  if (props.isButtonPressed) {
    players = {};
  }

  if (props.refreshed) {
    players = {};
    dataCleaner(false)
  }

  return (
    <View style={styles.ratingContainer}>
      <TextInput
        placeholder="Player Name"
        onChangeText={nameHandler}
        value={playerName}
        style={styles.inputText}
      />
      <View style={styles.pickerContainer}>
        <Picker
          style={styles.pickerStyle}
          mode="dropdown"
          selectedValue={abilityValue}
          onValueChange={abilityPicker}
        >
          <Picker.Item label="Ability" value={null} />
          <Picker.Item label="1" value={1} />
          <Picker.Item label="2" value={2} />
          <Picker.Item label="3" value={3} />
          <Picker.Item label="4" value={4} />
          <Picker.Item label="5" value={5} />
          <Picker.Item label="6" value={6} />
          <Picker.Item label="7" value={7} />
          <Picker.Item label="8" value={8} />
          <Picker.Item label="9" value={9} />
          <Picker.Item label="10" value={10} />
        </Picker>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  ratingContainer: {
    paddingTop: 8,
    width: "80%",
    height: 40,
    flexDirection: "row",
    justifyContent: "space-between",
    padding: 0,
  },
  inputText: {
    flex: 3,
    paddingLeft: 4,
    fontSize: 16,
    borderColor: "#1b411f",
    backgroundColor: "white",
    borderRadius: 8,
    borderWidth: 2,
  },
  pickerContainer: {
    alignItems: "center",
    justifyContent: "center",
    borderColor: "#1b411f",
    backgroundColor: "white",
    borderRadius: 8,
    borderWidth: 2,
  },
  pickerStyle: {
    width: 110,
  },
});
