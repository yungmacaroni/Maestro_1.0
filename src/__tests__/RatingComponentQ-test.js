import React from 'react';
import renderer from 'react-test-renderer';
import RatingComponentQ from '../components/RatingComponentQ';

test('renders correctly', () => {
  const tree = renderer.create(<RatingComponentQ />).toJSON();
  expect(tree).toMatchSnapshot();
});