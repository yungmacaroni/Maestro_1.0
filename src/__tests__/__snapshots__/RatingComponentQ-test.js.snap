// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "space-between",
      "padding": 0,
      "paddingTop": 8,
      "width": "80%",
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="Player Name"
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "#1b411f",
        "borderRadius": 8,
        "borderWidth": 2,
        "flex": 3,
        "fontSize": 16,
        "paddingLeft": 4,
      }
    }
    value=""
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "#1b411f",
        "borderRadius": 8,
        "borderWidth": 2,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "width": 110,
        }
      }
    >
      <RNCPicker
        items={
          Array [
            Object {
              "label": "Ability",
              "testID": undefined,
              "textColor": undefined,
              "value": null,
            },
            Object {
              "label": "1",
              "testID": undefined,
              "textColor": undefined,
              "value": 1,
            },
            Object {
              "label": "2",
              "testID": undefined,
              "textColor": undefined,
              "value": 2,
            },
            Object {
              "label": "3",
              "testID": undefined,
              "textColor": undefined,
              "value": 3,
            },
            Object {
              "label": "4",
              "testID": undefined,
              "textColor": undefined,
              "value": 4,
            },
            Object {
              "label": "5",
              "testID": undefined,
              "textColor": undefined,
              "value": 5,
            },
            Object {
              "label": "6",
              "testID": undefined,
              "textColor": undefined,
              "value": 6,
            },
            Object {
              "label": "7",
              "testID": undefined,
              "textColor": undefined,
              "value": 7,
            },
            Object {
              "label": "8",
              "testID": undefined,
              "textColor": undefined,
              "value": 8,
            },
            Object {
              "label": "9",
              "testID": undefined,
              "textColor": undefined,
              "value": 9,
            },
            Object {
              "label": "10",
              "testID": undefined,
              "textColor": undefined,
              "value": 10,
            },
          ]
        }
        numberOfLines={1}
        onChange={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;
